src/application/App.tsx

```diff
@@ -74,22 +74,28 @@ export const App = () => {
     devtoolsMachine.provide({
       actions: {
         resetStore: () => {
-          apolloClient.resetStore().catch(() => {});
+          apolloClient.clearStore().catch(() => {});
         },
       },
     })
   );
-  const { data, client: apolloClient } = useQuery(APP_QUERY, {
-    errorPolicy: "all",
-  });
+  const {
+    data,
+    client: apolloClient,
+    refetch,
+  } = useQuery(APP_QUERY, { errorPolicy: "all" });
 
   useActorEvent("connectToDevtools", () => {
     send({ type: "connect" });
   });
 
-  useActorEvent("registerClient", (message) => {
+  useActorEvent("registerClient", () => {
     send({ type: "connect" });
-    addClient(message.payload);
```

Comment:
I updated `addClient` to use `client.writeQuery` to run a `broadcastQueries`, but unfortunately that didn't seem to have an effect on the mounted queries after `clearStore` was run.

This is definitely something we should look into fixing in the client as its easy for `useQuery` to get out-of-sync with the cache after a `clearStore` has been called.